### ISSUE TITLE
NullPointerException during lsp initialization

### ISSUE THREAD

**Issue by amergey** [fwcd/kotlin-language-server#277]
Hello, 

I am using kotlin lsp in eclipse (i followed this tutorial to integrate the lsp with lsp4e https://www.vogella.com/tutorials/EclipseLanguageServer/article.html) 

Moving to latest lsp version it stops working and I have following stack trace
```

[Info]	main      Connected to client
[Error]	async0    Internal error: java.lang.NullPointerException: folders must not be null
[Error]		at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
[Error]	java.util.concurrent.CompletionException: java.lang.NullPointerException: folders must not be null
[Error]		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[Error]		at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1702)
[Error]		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[Error]		at java.base/java.lang.Thread.run(Thread.java:834)
[Error]	Caused by: java.lang.NullPointerException: folders must not be null
[Error]		at org.javacs.kt.KotlinLanguageServer$initialize$1.invoke(KotlinLanguageServer.kt:98)
[Error]		at org.javacs.kt.KotlinLanguageServer$initialize$1.invoke(KotlinLanguageServer.kt:22)
[Error]		at org.javacs.kt.util.AsyncExecutorKt$sam$java_util_function_Supplier$0.get(AsyncExecutor.kt)
[Error]		... 3 more
[Error]		at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
[Error]		at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
```
It seems this occurs since https://github.com/fwcd/kotlin-language-server/commit/55fa0aa822a8272af8cc7c6069ffb73846a9cdb1
where **workspaceFolders** init param is used instead of **rootUri**

Above you can see what lsp4e is sending to the lsp when the failure occurs.

At this point it is not clear to me if the issue is on lsp4e side or kotlin lsp side. What do you think ?

```
{
	"jsonrpc": "2.0",
	"id": "1",
	"method": "initialize",
	"params": {
		"processId": 21149,
		"rootPath": "/home/arnaud-mergey/workspaces/runtime-Kotlin/test/",
		"rootUri": "file:///home/arnaud-mergey/workspaces/runtime-Kotlin/test/",
		"capabilities": {
			"workspace": {
				"applyEdit": true,
				"workspaceEdit": {
					"documentChanges": true,
					"resourceOperations": [
						"create",
						"delete",
						"rename"
					],
					"failureHandling": "undo"
				},
				"symbol": {
					"dynamicRegistration": true
				},
				"executeCommand": {
					"dynamicRegistration": true
				},
				"workspaceFolders": true
			},
			"textDocument": {
				"synchronization": {
					"willSave": true,
					"willSaveWaitUntil": true,
					"didSave": true
				},
				"completion": {
					"completionItem": {
						"snippetSupport": true,
						"documentationFormat": [
							"markdown",
							"plaintext"
						]
					}
				},
				"hover": {
					"contentFormat": [
						"markdown",
						"plaintext"
					]
				},
				"signatureHelp": {},
				"references": {},
				"documentHighlight": {},
				"documentSymbol": {
					"symbolKind": {
						"valueSet": [
							18,
							17,
							5,
							14,
							9,
							10,
							22,
							24,
							8,
							1,
							12,
							11,
							20,
							6,
							2,
							3,
							21,
							16,
							19,
							25,
							4,
							7,
							15,
							23,
							26,
							13
						]
					},
					"hierarchicalDocumentSymbolSupport": true
				},
				"formatting": {
					"dynamicRegistration": true
				},
				"rangeFormatting": {},
				"definition": {
					"linkSupport": true
				},
				"typeDefinition": {
					"linkSupport": true
				},
				"codeAction": {
					"codeActionLiteralSupport": {
						"codeActionKind": {
							"valueSet": [
								"quickfix",
								"refactor",
								"refactor.extract",
								"refactor.inline",
								"refactor.rewrite",
								"source",
								"source.organizeImports"
							]
						}
					},
					"dynamicRegistration": true
				},
				"codeLens": {},
				"documentLink": {},
				"colorProvider": {},
				"rename": {}
			}
		},
		"clientName": "Eclipse Platform",
		"trace": "off"
	}
}
```

**Comment by fwcd** [comment]
`rootUri` (and `rootPath`) are deprecated in favor of `workspaceFolders`, which apparently is not supported by LSP4E yet, so I think the best solution would be to use `rootUri` as a fallback if no `workspaceFolders` are set.

**Comment by rduplain** [comment]
Thanks for fixing this issue. For anyone following along, this is fixed `>1.1.1`, which means (as of today) you have to build/install from `HEAD` (or checkout fc1ce40adb0e3709c505d814f4c2b8a4b4eb069e, the commit linked to this issue).

**Comment by sonique6784** [comment]
hopefully we can get a 1.1.2 soon, to help with that.
